### DISCUSSION TITLE
向量化join为什么要求左侧算子为nullable

### DISCUSSION THREAD

**Discussion by big-doudou** [apache/doris#17171] [RFC]
doris版本:1.1.2
执行以下sql会出错：
1105 - errCode = 2, detailMessage = errCode = 2, detailMessage = errCode = 2, detailMessage = The slot (SlotDescriptor{id=15, parent=11, col=join_key, type=TINYINT, materialized=false, byteSize=0, byteOffset=-1, nullIndicatorByte=0, nullIndicatorBit=0, slotIdx=0}) could not be changed to nullable, Time: 0.024000s

SELECT * from
(select 1 as join_key from test where id in (SELECT id from dim where create_date in (SELECT max(create_date) from dim)))t1
LEFT JOIN 
(select 1 as join_key from test where id in (SELECT id from dim where create_date in (SELECT max(create_date) from dim)))t2
on t1.join_key=t2.join_key
LEFT JOIN 
(select 1 as join_key from test where id in (SELECT id from dim where create_date in (SELECT max(create_date) from dim)))t3
on t1.join_key=t3.join_key

我不确定这是否是一个bug

另外下面的pr，为什么要求列为nullable
https://github.com/apache/doris/pull/9954

**Comment by yagagagaga** [comment]
It would be better if you sent out your table DDL.

**Reply by big-doudou** [reply]
dim表DDL：
CREATE TABLE `dim` (
  `id` bigint(20) NULL DEFAULT "0" COMMENT "账户ID",
  `create_date` date NOT NULL DEFAULT "0000-01-01" COMMENT "时间日期"
) ENGINE=OLAP
UNIQUE KEY(`id`, `create_date`)
COMMENT "OLAP"
PARTITION BY RANGE(`create_date`)()
DISTRIBUTED BY HASH(`id`) BUCKETS 3
PROPERTIES (
"in_memory" = "false",
"storage_format" = "V2"
)


test表是一张普通的AGG表，join_key字段是不存在的:
CREATE TABLE `test` (
  `id` bigint NULL DEFAULT "0" COMMENT "账户ID",
	`create_date` date NOT NULL DEFAULT "0000-01-01" COMMENT "时间日期",
  `views` int(11) SUM NULL DEFAULT "0" COMMENT "曝光"
) ENGINE=OLAP
AGGREGATE KEY(`id`,`create_date`)
COMMENT "OLAP"
PARTITION BY RANGE(`create_date`)
()
DISTRIBUTED BY HASH(`id`) BUCKETS 18
PROPERTIES (

"in_memory" = "false",
"storage_format" = "V2"
);

**Reply by yagagagaga** [reply]
I test it in Doris 1.2.2-rc01 , everything work well. Maybe you can upgrade your Doris version.

**Reply by big-doudou** [reply]
3q~   
 Let me try~
